docs/topics/proc_starting-your-application-locally-in-debugging-mode.adoc

```diff
@@ -0,0 +1,31 @@
+
+[#starting-your-application-locally-in-debugging-mode_{context}]
+= Starting Your Application Locally in Debugging Mode
+
+One of the ways of debugging a Maven-based project is manually launching the application while specifying a debugging port, and subsequently connecting a remote debugger to that port.
+This method is applicable at least to the following deployments of the application:
+
+* When launching the application manually using the `mvn wildfly-swarm:run` goal.
+* When starting the application without waiting for the it to exit using the `mvn wildfly-swarm:start` goal.
+This is useful especially when performing integration testing.
+* When using the Arquillian adapter for {runtime}.
```

Comment:
Neither Spring Boot nor Vert.x provide an Arquillian adapter, AFAIK. And if they did, they most likely wouldn't use a system property called `swarm.xxx`, as described below. Also their Maven plugins aren't used like `mvn wildfly-swarm:*`. All in all, this topic is specific to WildFly Swarm.

Reply:
Of course, those are the sort of alterations I will have to make if we decide to make this common. For now, I felt hardcoding the values is fine.

Reply:
Yea sure. There's one Swarm-specific file that is fairly generic (I added a comment); on the other hand, this one is very Swarm-specific even though its name is generic.

Reply:
If the steps are the same (meaning you run a single command to start the app where you specify the port), we can make the file reasonably reusable through conditions and attributes. I just need to know the delta between runtimes. If it's indeed too big, I'll rename this file to WFS-specific only to reflect the fact.

Reply:
@tradej @inoxx03 I agree.  I we make this portion of the examples fit our common use case. Spring Boot could reuse some of this content.  Much of the content around enabling debug and setting environment variables for Java based runtimes could be made common/parameterized to allow us to add the runtime  specific changes needed.

Reply:
LGTM and we can get started with re-using the common portions. I may need to use different JAVA environment variables and add a usecase for testing different input parameters via JAVA_OPTIONS.